cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -375,13 +435,87 @@ writeLedgerState mOutFile qState@(SerialisedLedgerState serLedgerState) =
     Just (OutputFile fpath) ->
       handleIOExceptT (ShelleyQueryCmdWriteFileError . FileIOError fpath)
         $ LBS.writeFile fpath $ unSerialised serLedgerState
- where
-   decodeLedgerState
-     :: SerialisedLedgerState era
-     -> Either LBS.ByteString (LedgerState era)
-   decodeLedgerState (SerialisedLedgerState (Serialised ls)) =
-     first (const ls) (decodeFull ls)
 
+writeStakeSnapshot :: forall era ledgerera. ()
+  => ShelleyLedgerEra era ~ ledgerera
+  => Era.Era ledgerera
+  => FromCBOR (LedgerState era)
+  => Hash StakePoolKey
```

Comment:
It may be clearer to use the type synonym `PoolId` here

Reply:
Fixed!